Selection tool and slit tool fails during the update of properties

Dear Fitroom team,

After solution of the problem #7 and #8 it appeared that there are some more places to edit on this topic.
When I try to re-select new area in SelectTool or when I am trying to update SlitTool after the first loop is finished, some errors appear:
1) If I reselect something in selection tool, the problem is still present (I updated and reinstalled fitroom). So traceback:
```
An error occured during selectTool callback
Traceback (most recent call last):
  File "c:\users\blekhshtein\fitroom\fitroom\selection_tool.py", line 266, in line_select_callback
    self.update_action(xmin, xmax, ymin, ymax)
  File "c:\users\blekhshtein\fitroom\fitroom\selection_tool.py", line 222, in update_action
    self.fitroom.update([xmin, xcen, xmax], [ymin, ycen, ymax])
  File "c:\users\blekhshtein\fitroom\fitroom\room.py", line 149, in update
    self.slitTool.update_figure()  # in case
  File "c:\users\blekhshtein\fitroom\fitroom\slit_tool.py", line 154, in update_figure
    w = s._q['wavespace']
KeyError: 'wavespace'
Traceback (most recent call last):
  File "C:\Users\Blekhshtein\anaconda3\envs\radis-env\lib\site-packages\matplotlib\cbook\__init__.py", line 270, in process
    func(*args, **kwargs)
  File "C:\Users\Blekhshtein\anaconda3\envs\radis-env\lib\site-packages\matplotlib\widgets.py", line 1841, in release
    self._release(event)
  File "C:\Users\Blekhshtein\anaconda3\envs\radis-env\lib\site-packages\matplotlib\widgets.py", line 2396, in _release
    self.onselect(self.eventpress, self.eventrelease)
  File "c:\users\blekhshtein\fitroom\fitroom\selection_tool.py", line 266, in line_select_callback
    self.update_action(xmin, xmax, ymin, ymax)
  File "c:\users\blekhshtein\fitroom\fitroom\selection_tool.py", line 222, in update_action
    self.fitroom.update([xmin, xcen, xmax], [ymin, ycen, ymax])
  File "c:\users\blekhshtein\fitroom\fitroom\room.py", line 149, in update
    self.slitTool.update_figure()  # in case
  File "c:\users\blekhshtein\fitroom\fitroom\slit_tool.py", line 154, in update_figure
    w = s._q['wavespace']
KeyError: 'wavespace'
```
Can this be caused by incompatible units?
2. When I change slit through the slitTool error also appears, saying that there is nothing for slit application, Traceback:
```
Traceback (most recent call last):
  File "C:\Users\Blekhshtein\anaconda3\envs\radis-env\lib\site-packages\matplotlib\cbook\__init__.py", line 270, in process
    func(*args, **kwargs)
  File "C:\Users\Blekhshtein\anaconda3\envs\radis-env\lib\site-packages\matplotlib\widgets.py", line 530, in <lambda>
    return self._observers.connect('changed', lambda val: func(val))
  File "c:\users\blekhshtein\fitroom\fitroom\slit_tool.py", line 345, in update_slider
    slabsTool.update_slit()
  File "c:\users\blekhshtein\fitroom\fitroom\multislab_tool.py", line 159, in update_slit
    s.apply_slit(slit_function, **slit_options)
  File "c:\users\blekhshtein\radis\radis\spectrum\spectrum.py", line 2174, in apply_slit
    raise AssertionError(
AssertionError: No variables to apply slit on. Variable names to be convolved should end with _noslit
```

--- a/fitroom/grid3x3_tool.py
+++ b/fitroom/grid3x3_tool.py
@@ -265,9 +265,8 @@ class Grid3x3():
             return
 
         # calculate residuals
-        s = s.take(plotquantity)
         res = get_residual(s)
-        w, I = s.get(plotquantity, wunit='nm', Iunit=unit)
+        w, I = s.take(plotquantity).get(plotquantity, wunit='nm', Iunit=unit)
 
         ydata = norm_on(w, I) if normalize else I
         # get normalizing factor (to print it)

--- a/fitroom/multislab_tool.py
+++ b/fitroom/multislab_tool.py
@@ -493,7 +493,7 @@ class MultiSlabPlot():
         slit_options = self.fitroom.solver.slit_options
 
         # Central axe: model vs experiment
-        w, I = s.get(plotquantity, Iunit=unit, wunit=self.wunit)
+        w, I = s.take(plotquantity).get(plotquantity, Iunit=unit, wunit=self.wunit)
         ydata = norm_on(w, I) if normalize else I
         try:
             line3cent[1].set_data(w, ydata)

--- a/fitroom/solver.py
+++ b/fitroom/solver.py
@@ -13,6 +13,7 @@ get the correct slab input, then calls the appropriate functions in neq.spec eng
 
 from __future__ import absolute_import
 from __future__ import print_function
+from typing import Union
 import numpy as np
 from scipy.interpolate import splev, splrep
 from warnings import warn
@@ -155,7 +156,7 @@ class SlabsConfigSolver():
         ''' Triggered on connection to FitRoom '''
         self.fitroom = fitroom         # type: FitRoom
 
-    def get_residual(self, s, normalize=False, normalize_how='max'):
+    def get_residual(self, s: Spectrum, normalize=False, normalize_how='max'):
         ''' Returns difference between experimental and simulated spectra
         By default, uses :func:`~radis.spectrum.compare.get_residual` function
         You can change the residual by overriding this function. 
@@ -229,7 +230,7 @@ class SlabsConfigSolver():
 #
 #        return error
 
-    def calc_slabs(self, **slabsconfig):
+    def calc_slabs(self, **slabsconfig) -> Union[Spectrum, dict, dict]:
         '''
         Parameters
         ----------

